### PR TITLE
CTDA-1669: Add metrics relating to PPNS

### PR DESCRIPTION
**Pull request by drnaylor-oc**
Related to https://github.com/hmrc/transit-movements-trader-at-destination/pull/284

### CODE REVIEW

**Review by RobertLaverick — APPROVED** [review]

**Review by RobertLaverick — APPROVED** [review]

File: app/audit/AuditService.scala [hunk 1]
```diff
@@ -36,9 +38,19 @@ class AuditService @Inject()(auditConnector: AuditConnector, messageTranslator:
...
+  def auditEvent(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType)(implicit hc: HeaderCarrier): Unit =
+    auditEventWithBox(auditType, customerId, message, channel, None)
+
+  def auditEventWithBox(auditType: AuditType, customerId: Ior[TURN, EORINumber], message: Message, channel: ChannelType, boxOpt: Option[Box])(
+    implicit hc: HeaderCarrier): Unit = {
     val details = AuthenticatedAuditDetails(channel, customerId, messageTranslator.translate(toJson(message.message)))
-    auditConnector.sendExplicitAudit(auditType.toString, details)
+    boxOpt match {
+      case Some(box) =>
+        val detailsWithBox = Json.toJson(details).as[JsObject] + ("boxId" -> JsString(box.boxId.value))
```
Comment (RobertLaverick) [inline app/audit/AuditService.scala:49]: I'd rather you used a new auditing case class (AuthenticatedAuditDetailsWithBox?) rather than bodging the Json.

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

File: app/audit/AuditService.scala (hunk 1 above)
Comment (drnaylor-oc) [reply app/audit/AuditService.scala:49]: Yeah, that's fair enough, I'll do that.

File: app/audit/AuditDetails.scala [hunk 2]
```diff
@@ -18,20 +18,30 @@ package audit
...
+}
+
 sealed abstract class AuditDetails {
   def channel: ChannelType
   def customerId: String
   def json: JsObject
 }
 
-case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject) extends AuditDetails {
+case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject, boxId: Option[BoxId]) extends AuditDetails {
```
Comment (RobertLaverick) [inline app/audit/AuditDetails.scala:44]: Have you considered that boxId could default to None?

File: app/audit/AuditDetails.scala [hunk 3]
```diff
@@ -51,11 +61,11 @@ case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN
...
-    Json.obj(
-      "channel"       -> details.channel,
-      "customerId"    -> details.customerId,
-      "enrolmentType" -> details.enrolmentType
-    ) ++ details.json
+    val json: JsObject = AuditDetails.writeAuditDetails(details) ++ Json.obj("enrolmentType" -> details.enrolmentType)
+    details.boxId match {
+      case Some(boxId) => json ++ Json.obj("boxId" -> boxId)
+      case None        => json
+    }
```
Comment (RobertLaverick) [inline app/audit/AuditDetails.scala:68]: can map on BoxId (Or fold? One of the two)

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**